src/hotspot/share/oops/symbol.hpp

```diff
@@ -155,7 +155,7 @@ class Symbol : public MetaspaceObj {
   // Returns the largest size symbol we can safely hold.
   static int max_length() { return max_symbol_length; }
   unsigned identity_hash() const {
-    unsigned addr_bits = (unsigned)((uintptr_t)this >> (LogMinObjAlignmentInBytes + 3));
+    unsigned addr_bits = (unsigned)((uintptr_t)this >> (LogBytesPerWord + 3));
```

Comment:
[pre-existing] I'm puzzled by the `+3` in this line.  I'm guessing this is `log2(sizeof(Symbol))`?

Reply:
Oh that might be a good guess.  I thought it was just some way to randomize the address some more.  @yminqi or @iklam  would know since it was added with JDK-8130115.

Reply:
In 32 bits, LogByesPerWord is 2, but in LogMinObjAlignmentInBytes is still 3:

Reply:
This address_bits here is for using 'this' to join the calculation of identity_hash. I don't think it matters which bits are used here, but we need keep consistency with SA. If you changed here, you need to change Symbol.java too. Note in 32 bits, LogBytesPerWord is different from LogMinObjAlignmentInBytes (it is 3 for both 32 and 64 its).

Reply:
I agree that the `+3` can be replaced with `log2(sizeof(Symbol))`. If I remember correctly, the intention was to avoid getting the same value for `(((uintptr_t)this) >> LogBytesPerWord) & 0x07)`.

However, this may not be necessary. The Symbols are of variable sizes (the string body is allocated as part of the Symbol). I write a program to analyze the distribution of the above expression for 18487 Symbols allocated for running a HelloWorld Java program. While it's not perfectly distributed, it may be good enough.

Perhaps we can get rid of the `+ 3` or `+ log2(sizeof(Symbol))` in a separate RFE?

```
0: +++++++
1: +++++++++++++++++++++++++++++++++++++++
2: ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
3: ++++++++++++++++++++++++++++++++++++++++++++++
4: ++++++++++++++++++++++++++++++
5: ++++++++++++++++++++++
6: +++++++++++++++
7: ++++++++++

0:  571
1: 3152
2: 4659
3: 3721
4: 2467
5: 1801
6: 1276
7:  838
```

Reply:
Thanks Yumin for pointing out the SA dependencies.  It turns out that this function isn't ultimately used by the SA anymore, but I've made the changes anyway.

Reply:
I'd be happy if someone changed the 3 to log2(sizeof(Symbol) in another RFE since I'm having trouble commenting why this is important.